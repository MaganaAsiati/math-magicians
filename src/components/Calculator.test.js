import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from './Calculator';

test('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('To test the functionality of calculator component', () => {
  test('should Show 27 if buttons 2,1,+ and 6 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: '+' }));
    fireEvent.click(screen.getByRole('button', { name: '6' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('27')).toBeTruthy();
  });

  test('should Show 0.25 if buttons .,2 and 5 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '.' }));
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '5' }));

    expect(screen.getByText('0.25')).toBeTruthy();
  });

  test('should Show 12 if buttons 4,x and 3 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '4' }));
    fireEvent.click(screen.getByRole('button', { name: 'x' }));
    fireEvent.click(screen.getByRole('button', { name: '3' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('12')).toBeTruthy();
  });

  test('should Show -91 if buttons 9,1 and +/- are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '9' }));
    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: '+/-' }));

    expect(screen.getByText('-91')).toBeTruthy();
  });


  test('should Show 11 if buttons 1,2,1,+,1 and 1 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: 'รท' }));
    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: '1' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('11')).toBeTruthy();
  });

  test('should Show 2 if buttons 5,0,%, and 6 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '5' }));
    fireEvent.click(screen.getByRole('button', { name: '0' }));
    fireEvent.click(screen.getByRole('button', { name: '%' }));
    fireEvent.click(screen.getByRole('button', { name: '6' }));

    expect(screen.getByText('2')).toBeTruthy();
  });

  test('should Show -3 if buttons 2,4,+/-,%, and 7 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '4' }));
    fireEvent.click(screen.getByRole('button', { name: '+/-' }));
    fireEvent.click(screen.getByRole('button', { name: '%' }));
    fireEvent.click(screen.getByRole('button', { name: '7' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('-3')).toBeTruthy();
  });

  test('should Show 1.75 if buttons 2,.,2,5,-,. and 5 are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '.' }));
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '5' }));
    fireEvent.click(screen.getByRole('button', { name: '-' }));
    fireEvent.click(screen.getByRole('button', { name: '.' }));
    fireEvent.click(screen.getByRole('button', { name: '5' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('1.75')).toBeTruthy();
  });

  test('should Show -7 if buttons 3,+/-,+,4 and +/- are clicked', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByRole('button', { name: '3' }));
    fireEvent.click(screen.getByRole('button', { name: '+/-' }));
    fireEvent.click(screen.getByRole('button', { name: '+' }));
    fireEvent.click(screen.getByRole('button', { name: '4' }));
    fireEvent.click(screen.getByRole('button', { name: '+/-' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));

    expect(screen.getByText('-7')).toBeTruthy();
  });
});
