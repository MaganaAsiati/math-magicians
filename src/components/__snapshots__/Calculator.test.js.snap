// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="calcTitle"
  >
    Let's do some math!
  </h1>,
  <div
    className="Calculator"
  >
    <span
      className="Display"
    >
      0
    </span>
    <section
      className="Keypad"
    >
      <button
        className="Button AC Gray"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="Button Inverter Gray"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="Button Modulo Gray"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="Button Divide Orange"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className="Button Seven Gray"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="Button Eight Gray"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="Button Nine Gray"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="Button Multiply Orange"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className="Button Four Gray"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="Button Five Gray"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="Button Six Gray"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="Button Minus Orange"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="Button One Gray"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="Button Two Gray"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="Button Three Gray"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="Button Plus Orange"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="Button Zero Gray"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="Button Dot Gray"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="Button Equal Orange"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </section>
  </div>,
]
`;
